docs/source/reference/template.rst

```diff
@@ -0,0 +1,75 @@
+
+.. _template:
+
+########
+Template
+########
+
+********
+The task
+********
+
+Here you should add a description of your task. For example:
+Classification is the task of assigning one of a number of classes to each data point.
+The :class:`~flash.template.TemplateSKLearnClassifier` is a :class:`~flash.core.model.Task` for classifying the datasets included with scikit-learn.
+
+------
+
+*********
+Inference
+*********
+
+Here, you should add a short intro to your predict example, and then use ``literalinclude`` to add it.
+
+.. note:: We skip the first 14 lines as they are just the copyright notice.
+
+Our predict example uses a model pre-trained on the iris data.
+
+.. literalinclude:: ../../../flash_examples/predict/template.py
+    :language: python
+    :lines: 14-
+
+For more advanced inference options, see :ref:`predictions`.
+
+------
+
+********
+Training
+********
+
+In this section, we briefly describe the data, and then ``literalinclude`` our finetuning example.
+
+Now we'll train on Fisher's classic iris data.
+It contains 150 records with four features (sepal length, sepal width, petal length, and petal width) in three classes (species of Iris: setosa, virginica and versicolor).
```

Comment:
Include link to images to make your description better.

Reply:
This is just tabular data, so I'm not sure what images we would show here